controllers/scopetemplate_controller.go

```diff
@@ -230,3 +273,20 @@ func (r *ScopeTemplateReconciler) deleteClusterRoles(ctx context.Context, listOp
 	}
 	return nil
 }
+
+func (r *ScopeTemplateReconciler) updateScopeTemplateCondition(ctx context.Context, st *operatorsv1.ScopeTemplate, condition metav1.Condition) {
+	// Get the latest instance of the ScopeTemplate in case it has been updated
+	tempSt := &operatorsv1.ScopeTemplate{}
+	err := r.Client.Get(ctx, client.ObjectKeyFromObject(st), tempSt)
+	if err != nil {
+		log.Log.Error(err, "failed to get the latest version of the ScopeTemplate to update the ScopeTemplate status")
+		return
+	}
```

Comment:
This is actually counterproductive. It seems like the right thing to do to avoid conflict errors from the apiserver, but the conflict error is essentially telling you: "the status you're setting is out of date"

The best thing to do is to try and set the status on the version of the scopeTemplate you got from the cache at the beginning of the reconcile, and if that fails for whatever reason, just return that error from `Reconcile` to force a re-reconcile.

Reply:
Makes sense. I actually was looking into #7 and doing some research on server side apply. Since we want to go that direction anyways I am thinking to go back and have it use that instead.

I could still have it return any errors and re-reconcile if necessary.

WDYT?

Reply:
Sgtm!